docs/proposals/0000-support-interchangeable-build-and-run-implementations.md

```diff
@@ -0,0 +1,248 @@
+  indicate their status as an external interface. We may even want to reclaim the `workloads.cloudfoundry.org` group for
+  this purpose and choose another for the existing resources.
+
+- run workload reconciler - similar to the build workload reconciler, but for `RunWorkload` resources.
+
+- the `CFBuild` reconciler: a component of `korifi-controllers` that creates `BuildWorkload` resources and monitors
+  their status to keep the `CFBuild` and/or `CFDroplet` resources up to date.
+
+- the `CFProcess` reconciler: a component of `korifi-controllers` that creates `RunWorkload` resources and monitors
+  their status to keep `CFApp` and/or `CFProcess` resources up to date.
+
+- the `CFServiceBinding` reconciler: a new component of `korifi-controllers` that creates and updates a `Secret`
+  resource for each app, containing the `VCAP_SERVICES` environment variable. This secret is referenced by both
+  the `BuildWorkload` and `RunWorkload` resources.
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+These diagrams demonstrate the various k8s resources and how they're related:
+
+![build-flow.png](assets/build-flow.png)
+
+![run-flow.png](assets/run-flow.png)
+
+Here are some (DRAFT) sample resources:
+
+```yaml
+apiVersion: TBD.cloudfoundry.org/v1alpha1
+kind: BuildWorkload
+metadata:
+  name: my-build-guid
+  namespace: my-space-guid
+spec:
+  env:
+    - name: VCAP_SERVICES
+      valueFrom:
+        secretKeyRef:
+          key: VCAP_SERVICES
+          name: vcap-services-secret
+  buildpacks: [] # Use empty to indicate "default"?
+  stack: io.buildpacks.stacks.bionic
+  source:
+    registry:
+      image: localregistry-docker-registry.default.svc.cluster.local:30050/kpack/packages/eb60d1d0-5de1-4051-ac05-a4047f2b191a@sha256:987d6b085314a16ba5566fd0821d5a2f59e1a54ea11b8c0e2cd09309482d0481
+      imagePullSecrets:
+        - name: image-registry-credentials
+status:
+  conditions:
+  - status: 'False'
+    type: Staging
+  - status: 'True'
+    type: Succeeded
+  droplet:
+    ports: []
+    processTypes:
+      - command: node /workspace/server.js
+        type: web
+    registry:
+      image: localregistry-docker-registry.default.svc.cluster.local:30050/korifi-controllers/kpack/images/dbcee519-d239-4272-a264-3c1ce36a3e0a@sha256:6cee35c7d3ddc688bc45745e7c54432063221f71c6a092a649dcdff17a028c90
+      imagePullSecrets:
+        - name: image-registry-credentials
+    stack: io.buildpacks.stacks.bionic
+```
+
+```yaml
+---
+apiVersion: TBD.cloudfoundry.org/v1alpha1
+kind: RunWorkload
+metadata:
+  name: my-proc-guid
+  namespace: my-space-guid
+spec:
+  command:
+    - /cnb/lifecycle/launcher
+    - node /workspace/server.js
+  
+  resources:
+    limits:
+      ephemeral-storage: 1024M
+      memory: 1024M
+    requests:
+      cpu: '0'
+      memory: 1024M
+      
+  envFrom:
+    - secretRef:
+        name: user-specified-env-vars
+        
+  env:
```

Comment:
This looks like the K8s `EnvVar` type which is nice because it's something we want to add to the Eirini LRP anyway.

* https://github.com/cloudfoundry/korifi/issues/918
* https://github.com/cloudfoundry/korifi/issues/917

Reply:
That's exactly what it is. I was thinking that we could just accept `env` and `envFrom` like the pod spec does, rather than having our own different representation for this information. In general, I'm thinking that we have something pod-spec-like, using https://github.com/knative/specs/blob/main/specs/serving/knative-api-specification-1.0.md#revisiontemplatespec as an example.